File: MyRegistrationPage/app/src/main/java/com/example/myregistrationpage/MainActivity.java
package com.example.myregistrationpage;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.RadioGroup;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    EditText firstName;
    EditText lastName;
    EditText email;
    private RadioGroup rGoup;
    private TextView gender;

    @Override
    protected void onCreate(Bundle savedInstanceState) {

        super.onCreate(savedInstanceState);
    }

    public void startProgress(View view) {
        Intent startProgress=new Intent(this,myprogress.class);
        startService(startProgress);
    }
}

File: Book/TimerApp/app/src/main/java/com/example/timerapp/MainActivity.kt
package com.example.timerapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import com.example.timerapp.databinding.ActivityMainBinding
import java.io.File
import java.util.Timer
import kotlin.concurrent.fixedRateTimer

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        findViewById<Button>(R.id.button).setOnClickListener {
        finish()
        }
    }

    override fun onDestroy() {
        super.onDestroy()
        var text:EditText=findViewById(R.id.text_view_id_referance)
        File(filesDir,"dataFile.text").writeText(text.toString())
    }


}



